File: Java/toyProblems/very-easy/solved/shuffle-string.java
// Leetcode #1528
// https://leetcode.com/problems/shuffle-string/

// 1/3/2022

class Solution {

  public String restoreString(String s, int[] indices) {
    String[] resultArr = new String[indices.length];
    String[] arrOfStrs = new String[indices.length];
    arrOfStrs = s.split("", 0);

    for (int i = 0; i < indices.length; i++) {
      resultArr[indices[i]] = arrOfStrs[i];
    }

    String resultStr = String.join("", resultArr);
    return resultStr;
  }
}

File: javascript/solved/very-easy/shuffleString.js
// Leet Code #5128
// https://leetcode.com/problems/shuffle-string/

var restoreString = function(s, indices) {
    let obj = {};
    
    for (let i = 0; i < indices.length; i++) {
        obj[indices[i]] = s[i];
    }
    
    return Object.values(obj).join('') 
};
